testing/python/fixtures.py

```diff
@@ -4236,6 +4364,25 @@ def test_suffix(fix_combined):
     assert result.ret == 0
 
 
+class TestFinalizerOnlyAddedOnce:
+    @pytest.fixture(scope="class", autouse=True)
+    def a(self):
+        pass
+
+    @pytest.fixture(scope="class", autouse=True)
+    def b(self, a):
+        pass
+
+    def test_a_will_finalize_b(self, request):
+        a = request._get_active_fixturedef("a")
+        b = request._get_active_fixturedef("b")
+        assert b._will_be_finalized_by_fixture(a)
+
+    def test_a_only_finishes_one(self, request):
```

Comment:
Not sure about this one but I think you meant

```suggestion
    def test_a_only_finishes_once(self, request):
```